File: tools/apk_self/src/com/aigc/dashboard/MainActivity.java
package com.aigc.dashboard;

import android.app.Activity;
import android.graphics.Color;
import android.os.Bundle;
import android.view.Gravity;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.FrameLayout;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

public class MainActivity extends Activity {
    private WebView web;
    private FrameLayout overlay;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        try {
            FrameLayout root = new FrameLayout(this);
            web = new WebView(this);
            web.getSettings().setJavaScriptEnabled(true);
            web.getSettings().setDomStorageEnabled(true);
            web.setWebViewClient(new WebViewClient() {
                @Override
                public void onPageFinished(WebView view, String url) {
                    if (overlay != null) {
                        overlay.postDelayed(new Runnable() {
                            @Override
                            public void run() {
                                overlay.setVisibility(android.view.View.GONE);
                            }
                        }, 1000);
                    }
                }
            });
            root.addView(web, new FrameLayout.LayoutParams(
                    FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.MATCH_PARENT));

            overlay = new FrameLayout(this);
            overlay.setBackgroundColor(Color.rgb(11, 18, 32));
            LinearLayout box = new LinearLayout(this);
            box.setOrientation(LinearLayout.VERTICAL);
            box.setGravity(Gravity.CENTER);
            ProgressBar pb = new ProgressBar(this);
            TextView status = new TextView(this);
            status.setText("正在检查配置并连接数据...\n（首次加载约 3~10 秒）");
            status.setTextColor(Color.rgb(148, 163, 184));
            box.addView(pb);
            box.addView(status);
            FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(
                    FrameLayout.LayoutParams.WRAP_CONTENT, FrameLayout.LayoutParams.WRAP_CONTENT);
            lp.gravity = Gravity.CENTER;
            overlay.addView(box, lp);
            root.addView(overlay, new FrameLayout.LayoutParams(
                    FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.MATCH_PARENT));

            setContentView(root);
            new LocalServer(readAsset("dashboard.html"), readAsset("posthog_key.txt")).start();
            web.loadUrl("http://127.0.0.1:9876/");
        } catch (Exception e) {
            try {
                setContentView(new TextView(this) {{
                    setText("启动失败：" + e);
                    setTextColor(Color.WHITE);
                }});
            } catch (Exception ignore) {
            }
        }
    }

    private String readAsset(String name) {
        try {
            java.io.InputStream is = getAssets().open(name);
            java.io.ByteArrayOutputStream bos = new java.io.ByteArrayOutputStream();
            byte[] buf = new byte[8192];
            int n;
            while ((n = is.read(buf)) > 0) {
                bos.write(buf, 0, n);
            }
            is.close();
            return bos.toString("UTF-8");
        } catch (Exception e) {
            return "<html><body>assets error: " + e + "</body></html>";
        }
    }

    @Override
    public void onBackPressed() {
        if (web.canGoBack()) {
            web.goBack();
        } else {
            super.onBackPressed();
        }
    }
}

File: tools/android_webview/app/src/main/java/com/aigc/dashboard/MainActivity.kt
package com.aigc.dashboard

import android.annotation.SuppressLint
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Button
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    // ===== 打包前改这里：填你电脑的局域网地址，例如 "http://192.168.1.100:8765" =====
    // 填了之后，APK 装到手机，打开就直接显示数据，无需任何设置。
    // 留空则首次打开会出现一个地址输入框（输入一次，以后自动记住）。
    private val DEFAULT_URL = ""

    private lateinit var web: WebView
    private lateinit var urlBar: LinearLayout
    private lateinit var urlInput: EditText

    private val prefs by lazy { getSharedPreferences("dashboard", MODE_PRIVATE) }

    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val root = LinearLayout(this)
        root.orientation = LinearLayout.VERTICAL

        urlBar = LinearLayout(this)
        urlBar.orientation = LinearLayout.HORIZONTAL
        urlBar.setPadding(10, 10, 10, 10)
        urlBar.setBackgroundColor(Color.rgb(15, 23, 42))
        urlInput = EditText(this)
        urlInput.hint = "输入电脑看板地址，如 192.168.1.100:8765"
        urlInput.setTextColor(Color.WHITE)
        urlInput.setHintTextColor(Color.rgb(148, 163, 184))
        val btn = Button(this)
        btn.text = "连接"
        urlBar.addView(
            urlInput,
            LinearLayout.LayoutParams(0, LinearLayout.LayoutParams.WRAP_CONTENT, 1f)
        )
        urlBar.addView(btn)
        root.addView(urlBar)

        web = WebView(this)
        web.settings.javaScriptEnabled = true
        web.settings.domStorageEnabled = true
        web.webViewClient = WebViewClient()
        root.addView(
            web,
            LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, 0, 1f)
        )
        setContentView(root)

        val saved = prefs.getString("url", "")
        when {
            !saved.isNullOrEmpty() -> openUrl(saved)
            DEFAULT_URL.isNotEmpty() -> openUrl(DEFAULT_URL)
            else -> btn.setOnClickListener { connect() }
        }
    }

    private fun connect() {
        var u = urlInput.text.toString().trim()
        if (u.isEmpty()) {
            Toast.makeText(this, "请输入地址", Toast.LENGTH_SHORT).show()
            return
        }
        if (!u.startsWith("http")) {
            u = "http://$u"
        }
        prefs.edit().putString("url", u).apply()
        openUrl(u)
    }

    private fun openUrl(u: String) {
        urlBar.visibility = View.GONE
        web.loadUrl(u)
    }

    @Deprecated("Deprecated in Java")
    override fun onBackPressed() {
        if (web.canGoBack()) web.goBack() else super.onBackPressed()
    }
}
